### PR TITLE
Profile page show post

### PR DESCRIPTION
**Pull request by michaelcychan**
The Profile page can show the user's own posts.

### CODE REVIEW

File: views/users/profile.hbs [hunk 1]
```diff
@@ -1,3 +1,39 @@
 <h1 class="title">Profile Page</h1>
 
-<p class="greeting">This profile page is about {{session.user.email}}</p>
\ No newline at end of file
+<p class="greeting">This profile page is about {{session.user.email}}</p>
+<div class="posts">
+  <ul class="posts">
+  {{#each posts}}
+    <div class="post-container">
```
Comment (fridayshoes) [inline views/users/profile.hbs:7]: Nice readable html

**Review by fridayshoes — APPROVED** [review]

**Review by kaykakaraka — COMMENTED** [review]
Code looks good! I would suggest a couple more tests to be added. But these could also be added after the merge

File: controllers/users.js [hunk 2]
```diff
@@ -17,9 +18,14 @@ const UsersController = {
...
+    const accountEmail = req.session.user.email;
+    Post.find({user: accountEmail}, (err, posts) => {
+      if (err) {
+        throw err;
+      }
+      posts.reverse() // reorders posts, so newest post is always at the top of the list
+      res.render("users/profile", {posts: posts, session: req.session});
+    });
   },
   ShowProfile: (req, res) => {
```
[the remark this answers is not in the document]
Comment (michaelcychan) [reply controllers/users.js:31]: It was supposed to show other people's profile page, not yet implemented.

File: cypress/integration/user_own_profile_page_spec.js [hunk 3]
```diff
@@ -18,4 +18,33 @@ describe("User's Own Profile Page", function() {
...
+      cy.get(".posts").should("contain", "Hello, world!");
+      cy.get(".posts").should("contain", "someone@example.com");
+      
+      //check content at profile page
+      cy.get('#view-self-profile').click();
+      cy.get(".posts").should("contain", "Hello, world!");
+      cy.get(".posts").should("contain", "someone@example.com");
+    });
+  });
+});
```
[the remark this answers is not in the document]
Comment (kaykakaraka) [reply cypress/integration/user_own_profile_page_spec.js:50]: and perhaps a final test.... to check that the posts are rendered in reverse order

File: controllers/users.js [hunk 4]
```diff
@@ -17,9 +18,14 @@ const UsersController = {
...
   SelfProfile: (req, res) => {
-    // console.log(req.session.user.email);
-    res.render("users/profile", {session: req.session})
-
+    const accountEmail = req.session.user.email;
+    Post.find({user: accountEmail}, (err, posts) => {
+      if (err) {
+        throw err;
+      }
+      posts.reverse() // reorders posts, so newest post is always at the top of the list
```
[the remark this answers is not in the document]
Comment (michaelcychan) [reply controllers/users.js:26]: Yes, the posts.reverse() in post controller would not affect the Post.find here in the users Controller, so we also need one here.

**Review by fridayshoes — APPROVED** [review]

**Pull request merged**